/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

option java_multiple_files = true;
package org.apache.airavata.mft.credential.stubs.gcs;

import "CredCommon.proto";

// This represents GCS Service account credentials. https://cloud.google.com/iam/docs/service-accounts
message GCSSecret {
    string secretId = 1;
    string projectId = 2;
    string privateKey = 3;
    string clientEmail = 4;
}

message GCSSecretGetRequest {
    string secretId = 1;
    org.apache.airavata.mft.common.AuthToken authzToken = 2;
}

message GCSSecretCreateRequest {
    string projectId = 1;
    string privateKey = 2;
    string clientEmail = 3;
    org.apache.airavata.mft.common.AuthToken authzToken = 4;
}

message GCSSecretUpdateRequest {
    string secretId = 1;
    string projectId = 2;
    string privateKey = 3;
    string clientEmail = 4;
    org.apache.airavata.mft.common.AuthToken authzToken = 5;
}

message GCSSecretUpdateResponse {
    string secretId = 1;
}

message GCSSecretDeleteRequest {
    string secretId = 1;
    org.apache.airavata.mft.common.AuthToken authzToken = 2;
}

message GCSSecretDeleteResponse {
    bool status = 1;
}